upload-api/functions/bridge.js

```diff
@@ -0,0 +1,156 @@
+import * as Sentry from '@sentry/serverless'
+import { invoke, DID } from '@ucanto/core'
+import { connect } from '@ucanto/client'
+import * as CAR from '@ucanto/transport/car'
+import * as HTTP from '@ucanto/transport/http'
+import { sha256 } from '@ucanto/core'
+import { ed25519 } from '@ucanto/principal'
+import { base64pad } from 'multiformats/bases/base64'
+import * as UCAN from "@ipld/dag-ucan"
+
+Sentry.AWSLambda.init({
+  environment: process.env.SST_STAGE,
+  dsn: process.env.SENTRY_DSN,
+  tracesSampleRate: 1.0,
+})
+
+/**
+ * 
+ * @param {Uint8Array} secret
+ * @returns 
+ */
+async function deriveSigner(secret) {
+  const { digest } = await sha256.digest(secret)
+  return await ed25519.Signer.derive(digest)
+}
+
+/**
+ * AWS HTTP Gateway handler for POST / with ucan invocation router.
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ *
+ * @param {import('aws-lambda').APIGatewayProxyEventV2} request
+ */
+async function handlerFn(request) {
+  try {
+    const { UPLOAD_API_DID, ACCESS_SERVICE_URL } = process.env
+
+    if (!UPLOAD_API_DID) {
+      return {
+        statusCode: 500,
+        body: 'UPLOAD_API_DID is not set'
+      }
+    }
+
+    if (!ACCESS_SERVICE_URL) {
+      return {
+        statusCode: 500,
+        body: 'ACCESS_SERVICE_URL is not set'
+      }
+    }
+
+    // parse headers
+    const authSecretHeader = request.headers['x-auth-secret']
+    if (!authSecretHeader) {
+      return {
+        statusCode: 401,
+        body: 'request has no x-auth-secret header'
+      }
+    }
+    const secret = base64pad.baseDecode(authSecretHeader)
+
+    const authorizationHeader = request.headers['authorization']
+    if (!authorizationHeader) {
+      return {
+        statusCode: 401,
+        body: 'request has no authorization header'
+      }
+    }
+    if (!authorizationHeader.startsWith('Bearer ')) {
+      return {
+        statusCode: 401,
+        body: 'authorization header must use the Bearer directive'
+      }
+    }
+    const jwt = authorizationHeader.replace('Bearer ', '')
+    const delegation = UCAN.parse(jwt)
+
+    // parse body
```

Comment:
Can we support content type negotiation here ? If content-type was application/json we can parse as such, but we could also accept `dag-cbor` in which case we should parse body as such.

Reply:
`accept` header can be used to decide how to format response.

Reply:
yea I think this would be great! also aligns with @gobengo's idea to use content negotiation to accept various iterations of the UCAN spec. 

I have a soft preference to pick one format to accept in this iteration, and follow pretty quickly with support for other content types - are json and dag-cbor the only two formats you'd imagine accepting here or is there a bigger list @Gozala ?